th2_data_services_lwdp/provider_api/grpc.py

```diff
@@ -191,36 +150,30 @@ def search_messages(
             start_timestamp=start_timestamp,
             end_timestamp=end_timestamp,
             result_count_limit=result_count_limit,
-            keep_open=keep_open,
             search_direction=search_direction,
-            filters=filters,
         )
 
         stream = self.__transform_streams(stream)
-        attached_events = BoolValue(value=attached_events)
         message_search_request = MessageSearchRequest(
             start_timestamp=basic_request.start_timestamp,
             end_timestamp=basic_request.end_timestamp,
             search_direction=basic_request.search_direction,
             result_count_limit=basic_request.result_count_limit,
-            keep_open=basic_request.keep_open,
-            filter=basic_request.filters,
-            attached_events=attached_events,
-            stream=stream,
             stream_pointer=stream_pointer,
+            stream=stream,
         )
 
         return self.__stub.searchMessages(message_search_request)
 
     @staticmethod
-    def __search_basic_checks(
+    def __search_basic_checks_message(
```

Comment:
It had common for events and messages checks.
You split it to two separate methods.
Now they have the same duplicate code.

What is the idea?

Reply:
Since message request has search direction and result count limit while events request doesn't, I thought it would be better to have different checks for each of them. It threw a missing argument error when I didn't pass in anything for those two variables in get_events methods. Is there better solution you could offer?

Reply:
You have duplicate code for some checks

The best solution here is to have common_checks and special checks for events and messages